SDCTA/src/components/auth/SignUpPage.tsx

```diff
@@ -0,0 +1,296 @@
+import { login } from "../../slices/loginSlice";
+import { useDispatch } from "react-redux";
+
+export const SignUpPage: React.FC = () => {
+  const [userDisplayName, setUserDisplayName] = useState("");
+  const [registerEmail, setRegisterEmail] = useState("");
+  const [registerPassword, setRegisterPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [agreedTerms, setAgreedTerms] = useState(false);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  let [inputError, setInputError] = useState({
+    unknownError: "",
+    passwordError: "",
+    confirmError: "",
+    emailError: "",
+    nameError: "",
+  });
+
+  const register = async (): Promise<void> => {
+    try {
+      inputError.unknownError = "";
+      inputError.passwordError = "";
+      inputError.confirmError = "";
+      inputError.emailError = "";
+      inputError.nameError = "";
+
+      let error = false;
+      // make sure passwords match.
+      if (!(confirmPassword === registerPassword)) {
+        inputError.confirmError = "Passwords do not match.";
+        error = true;
+      }
+
+      if (userDisplayName === "") {
+        inputError.nameError = "Please enter your full name.";
+        error = true;
+      }
+
+      if (!agreedTerms) {
+        inputError.unknownError =
+          "Must agree to the terms and services to register.";
+        error = true;
+      }
+
+      setInputError({
+        ...inputError,
+      });
+
+      if (!error) {
+        const userCredential = await createUserWithEmailAndPassword(
+          auth,
+          registerEmail,
+          registerPassword
+        );
+
+        await updateProfile(userCredential.user, {
+          displayName: userDisplayName,
+        });
+
+        await registerUser(userCredential);
+
+        dispatch(login());
+        navigate("/");
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        const errorMessage = signUpErrorHandler(error);
+
+        inputError = { ...inputError, ...errorMessage };
+        setInputError({ ...inputError });
+      }
+    }
+  };
+
+  const provider = new GoogleAuthProvider();
+  const auth_ = getAuth();
+
+  const loginWithGoogle = async () => {
+    signInWithRedirect(auth_, provider);
+    getRedirectResult(auth_)
+      .then((result) => {
+        // This gives you a Google Access Token. You can use it to access Google APIs.
+        // @ts-expect-error: Object is possibly 'null'.
+        const credential = GoogleAuthProvider.credentialFromResult(result);
+
+        // @ts-expect-error: Object is possibly 'null'.
+        const token = credential.accessToken;
+
+        // The signed-in user info.
+        // @ts-expect-error: Object is possibly 'null'.
+        const user = result.user;
+
+        // IdP data available using getAdditionalUserInfo(result)
+        // ...
+
+        navigate("/");
+      })
+      .catch((error) => {
+        // const errorCode = error.code;
```

Comment:
Do we still need this commented out code? If so, that's ok but lets try to remove it in a future auth PR